### DISCUSSION TITLE
Question regarding Search with expressions

### DISCUSSION THREAD

**Discussion by Grydpype** [ransome1/sleek#473] [Q&A] [answered]
Dear all,

I am unable to get the search with expressions to work; perhaps someone can advise me what I am doing wrong:

entering _due:2023-03-11_ (i.e. today's date) into the search field yields the desired results: all tasks due today.

However, according to the wiki page (https://github.com/ransome1/sleek/wiki/Filter-Expressions-for-Advanced-Search), I should also be able to write
_due = 2023-03-11
due = today
due == today
due < tomorrow_

none of which return anything.

What am I doing wrong? Is there an "activate search" setting somewhere I'm missing?

Using Sleek 1.3.1 on Mint 21.1

Kind regards,
Oliver.

**Comment by zerodat** [comment, marked as answer]
Hi, @Grydpype .  The wiki page is out of date.  Because of the change on Oct 31, 2022 to require a colon after a keyword, the correct syntax now is this:
```
due: = 2023-03-11
due: = today
due: == today
due: < tomorrow
```
I admit that the syntax is a little odd, but it is necessary so that plain-text search works as people expect.  I hope this works for you now that you know the correct syntax.

**Reply by Grydpype** [reply]
I have one further question here:

How would I search for tasks that do **not** have a due date? due: 0? NOT due:? 

Cheers,
Oliver.

**Reply by zerodat** [reply]
Yes, I think that `not due:` should do what you want, @Grydpype

**Reply by Grydpype** [reply]
Thanks for the quick reply.

However, it does look like the word 'not' is interpreted as a separate search criteria instead of a modifier for 'due:'; a first check resulted in me getting all tasks containing the (sub)string 'not' ...

Cheers,
Oliver.

**Reply by zerodat** [reply]
Did you actually try `not due:`?  If you type just `not` it will do a plain-text search, because `not` is not a valid search expression.  The `not` operator by itself is not a complete expression.  However, `not due:` is a complete expression and should match tasks which don't have a due date.

**Reply by Grydpype** [reply]
You were right; my mistake. Works as you said it would; thank you.

Cheers,
Oliver

**Comment by Grydpype** [comment]
hi @zerodat 

Thanks for the quick feedback; that solved it. Works fine now.

Cheers,
Oliver
